userland/examples/tutorials/xx_ble_adv/main.c

```diff
@@ -0,0 +1,93 @@
+#include <ble.h>
+#include <led.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <temperature.h>
+#include <timer.h>
+#include <tock.h>
+#include <unistd.h>
+
+const char* get_error(int err);
+
+/*******************************************************************************
+ * MAIN
+ ******************************************************************************/
+
+int main (void) {
+  int err;
+  printf("[Tutorial] BLE Advertising\n");
+
+  // BLE DATA FIELD
+  uint8_t addr[] = {1, 2, 3, 4, 5, 6};
+  uint16_t advertising_itv_ms = 300;
+  uint8_t device_name[]       = "TockOS";
+  uint16_t uuids[] = {0x1800, 0x1809};
+  uint8_t manufacturer_data[]     = {0x13, 0x37};
+  uint8_t fake_temperature_data[] = {0x00, 0x00};
+
+  // configure advertisement interval to 300ms
+  // configure LE only
+  // configure advertisement address as 1,2,3,4,5,6
+  err = ble_initialize(advertising_itv_ms,
+                       LE_LIMITED_DISCOVERABLE | BREDR_NOT_SUPPORTED,
+                       addr,
+                       sizeof(addr));
+  if (err < TOCK_SUCCESS) printf("ble_initialize, error: %s", get_error(err));
+
+  // configure device name as TockOS
+  err = ble_advertise_name(device_name, sizeof(device_name) - 1);
+  if (err < TOCK_SUCCESS) printf("ble_advertise_name, error: %s", get_error(err));
+
+  // configure list of UUIDs
+  err = ble_advertise_uuid16(uuids, sizeof(uuids));
+  if (err < TOCK_SUCCESS) printf("ble_advertise_uuid16, error: %s", get_error(err));
+
+  // configure manufacturer data
+  err = ble_advertise_manufacturer_specific_data(manufacturer_data, sizeof(manufacturer_data));
+  if (err < TOCK_SUCCESS) printf("ble_advertise_manufacturer_specific_data, error: %s", get_error(err));
+
+  // configure service data
+  err = ble_advertise_service_data(uuids[1], fake_temperature_data, sizeof(fake_temperature_data));
+  if (err < TOCK_SUCCESS) printf("ble_advertise_service_data, error: %s", get_error(err));
+
+  // start advertising
+  err = ble_start_advertising();
+  if (err < TOCK_SUCCESS) printf("ble_advertise_manufacturer_specific_data, error: %s", get_error(err));
+
+  return 0;
+}
+
+const char* get_error(int err) {
```

Comment:
Actually, this has already been written: https://github.com/helena-project/tock/blob/master/userland/libtock/tock.c#L178